azurelinuxagent/common/osutil/ubuntu.py

```diff
@@ -86,6 +86,22 @@ def __init__(self):
         super(Ubuntu18OSUtil, self).__init__()
         self.service_name = self.get_service_name()
 
+    def restart_if(self, ifname, retries=3, wait=5):
+        """
+        Restart systemd-networkd
+        """
+        retry_limit=retries+1
+        for attempt in range(1, retry_limit):
+            return_code=shellutil.run("systemctl restart systemd-networkd")
```

Comment:
Why is the logic for Ubuntu 18+ different than the one [already defined](https://github.com/Azure/WALinuxAgent/blob/develop/azurelinuxagent/common/osutil/ubuntu.py#L115-L130)? Doesn't the existing logic also achieve the same thing, or is its docstring incorrect?

Reply:
Have you checked #1544 ?
That is simply not working for Ubuntu 18+.